src/celengine/spheremesh.cpp

```diff
@@ -31,36 +31,27 @@ constexpr const int MinSlices = 3;
 SphereMesh::SphereMesh(const Eigen::Vector3f& size,
                        int _nRings, int _nSlices,
                        DisplacementMapFunc func,
-                       void* info)
+                       void* info) :
+    nRings(std::max(_nRings, MinRings)),
+    nSlices(std::max(_nSlices, MinSlices))
 {
-    createSphere(1.0f, std::max(MinRings, _nRings), std::max(MinSlices, _nSlices));
+    nVertices = nRings * (nSlices + 1);
+    nIndices = (nRings - 1) * (nSlices + 1) * 2;
+    vertices = std::unique_ptr<float[]>(new float[nVertices * 3]);
```

Comment:
make_unique?

Reply:
you also can move these assignments to constructor initializer block

Reply:
don't think that works with arrays

Reply:
``` 
   // Create a unique_ptr to an array of 5 elements
    std::unique_ptr<Vec3[]> v3 = std::make_unique<Vec3[]>(5);
 ```

Reply:
Ah right, it was arrays of *known size* at compile time...